### PR TITLE
chore: add warning text to v8 ContextualMenu example

### LINKED ISSUE #29127: [Bug]: ContextualMenu with SearchBox Accessibility Issue aria-required-children
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (16) x64 Intel(R) Xeon(R) W-2245 CPU @ 3.90GHz
    Memory: 96.78 GB / 127.59 GB
  Browsers:
    Edge: Spartan (1.1.0), Chromium (1.1.0), ChromiumDev (1.1.0)
    Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu

## Actual Behavior
The "ContextualMenu with custom rendered menu list that renders a search box to filter menu items" component is flagged by Accessibility Insights and reports the error seen in the logs.

Accessibility Insights Failure - Contextual Menu With Search Box

## Expected Behavior
All permutations of the `<ContextualMenu />` component pass accessibility insights without any reported issues.

## Notes
This seems related to issue #28343, but the difference between that issue and this one is that particular instance has both an `aria-hidden` element (the icon container) _and_ a `role=searchbox` element (the actual searchbox `input` element). The logs call out _both_ of these elements as being problematic.

### Logs

```shell
Title: WCAG 1.3.1: Ensures elements with an ARIA role that require child roles contain them (#id__294-menu)
Tags: Accessibility, WCAG 1.3.1, aria-required-children

Issue: Ensures elements with an ARIA role that require child roles contain them (aria-required-children - https://accessibilityinsights.io/info-examples/web/aria-required-children)

Target application: Fluent UI - Controls - React - ContextualMenu - https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu

Element path: #id__294-menu

Snippet: <div id="id__294-menu" class="ms-ContextualMenu-container container-247" tabindex="0" aria-labelledby="id__294" role="menu">

Related paths: 
  .iconContainer-411
  #SearchBox540

How to fix: 
Fix any of the following:
  Element has children which are not allowed: div[aria-hidden], [role=searchbox]
  Element uses aria-busy="true" while showing a loader

Environment: Chrome version 115.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.39.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
```

### Requested priority

Blocking

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

We have a ContextualMenu example showing how to customize it to add a text input before the menu items. Technically this is not an allowed parent/child role structure, though it more or less works in practice.

## New Behavior

We still have that example, but now with warning text about the error.

- Fixes #29127

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/a4410f44533c70f69f53c4985d2e72f4c41d874e'>a4410f44533c70f69f53c4985d2e72f4c41d874e</a></sub>
<div id="bundleSizeComment9423"/>

**Review by spmonahan — APPROVED** [review]

File: packages/react-examples/src/react/ContextualMenu/ContextualMenu.CustomMenuList.Example.tsx [hunk 1]
```diff
@@ -83,7 +84,17 @@ export const ContextualMenuWithCustomMenuListExample: React.FunctionComponent =
     [items, renderMenuList, onDismiss],
   );
 
-  return <DefaultButton text="Click for ContextualMenu" menuProps={menuProps} />;
+  return (
+    <>
+      <p>
+        <Text>
+          Warning: adding an input to a menu is not technically allowed, and will trigger errors in Accessibility
+          Insights and axe-core. Accessibility issues in practice are minimal, however.
```
Comment (mltejera) [inline packages/react-examples/src/react/ContextualMenu/ContextualMenu.CustomMenuList.Example.tsx:92]: Nit: Consider rephrasing in active voice:

"However, real world accessibility issues are minimal and or non-issues."

**Review by mltejera — APPROVED** [review]

File: packages/react-examples/src/react/ContextualMenu/ContextualMenu.CustomMenuList.Example.tsx (hunk 1 above)
Comment (smhigley) [inline packages/react-examples/src/react/ContextualMenu/ContextualMenu.CustomMenuList.Example.tsx:92]:
```suggestion
          Insights and axe-core. However, real world accessibility issues are minimal and not blocking.
```

Comment (smhigley) [reply packages/react-examples/src/react/ContextualMenu/ContextualMenu.CustomMenuList.Example.tsx]: done!

**Review by sopranopillow — APPROVED** [review]

**Pull request merged**